src/Symfony/Component/HttpClient/RetryableHttpClient.php

```diff
@@ -163,6 +164,14 @@ public function request(string $method, string $url, array $options = []): Respo
         });
     }
 
+    public function setLogger(LoggerInterface $logger): void
+    {
+        $this->logger = $logger;
```

Comment:
why this line?

Reply:
ah ok, got it

Reply:
`RetryableHttpClient` has it's own `$logger`, provided at construction
So, it act as a decorator, but it also has it's own agenda on logging

It is not required to have this line (for the purpose of this PR), but I think it's weird for a class implementing `LoggerAwareInterface` not to change it's own instance of logger when called...